### PR TITLE
Update Kubernetes core contributor slack channels

### PR DESCRIPTION
**Pull request by mrbobbytables**
This PR does 3 things:
- Renames kubernetes-contributors to kubernetes-new-contributors
- Creates kubernetes-maintainers
- Archives some unused channels

**Context:**
The kubernetes-contributors channel has become a place where people drop in and ask for support or requests from those completely new to OSS/K8s ignore the provided onboarding resources (join message + links to k8s.dev etc) and spam the channel or dm individuals with requests for direct 1:1 mentorship (sometimes riding the line of escalating to a CoC issue).

The channel has over 12k people in it, but our intended core audience, our k8s org members, have largely left, ignored, or muted the channel defeating its intended purpose.

To prevent this happening again to the new channel, the change will be backed up with a join message added to the kubernetes-maintainer channel stressing its intended purpose and statement of aggressive moderation. 

/hold
for review & any further discussion

### CODE REVIEW

**Comment by jeefy** [comment]
/lgtm

We don't currently lifecycle the description of a channel via Tempelis, so you should ensure that the description of the new `#kubernetes-maintainers` channel is clear. Maybe investigate some sort of auto-DM to someone new joining explaining the rules of it. 

Leaving the hold on for others to weigh in.

**Comment by mrbobbytables** [comment]
That is indeed the plan :+1: re:channel welcome message via dm. Will also update the one configured for the current channel

File: communication/slack-config/channels.yaml [hunk 1]
```diff
@@ -272,15 +272,19 @@ channels:
     archived: true
   - name: kubernetes-careers
   - name: kubernetes-client
-  - name: kubernetes-contributors
+  - name: kubernetes-org-members
+  - name: kubernetes-new-contributors
     id: C09R23FHP
```
Comment (Priyankasaggu11929) [inline communication/slack-config/channels.yaml:277]: Nit: need alpha sorting 
```suggestion
  - name: kubernetes-new-contributors
    id: C09R23FHP
  - name: kubernetes-org-members    
```

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]
Just left a small suggestion for alpha sorting.

/lgtm

File: communication/slack-config/channels.yaml (hunk 1 above)
Comment (mrbobbytables) [reply communication/slack-config/channels.yaml:277]: Ahh, thanks - I forgot to do that after changing it from maintainers to org members^^;;

**Comment by MadhavJivrajani** [comment]
/lgtm
Thank you!

**Review by palnabarun — APPROVED** [review]
/lgtm

Feel free to remove the hold when ready.

**Comment by DylanGraham** [comment]
/lgtm
🚀

**Review by nikhita — APPROVED** [review]
/lgtm
/approve 

:shipit:

**Comment by sftim** [comment]
@mrbobbytables you might like to update the PR description to reflect the change in channel name since the start of this PR

**Pull request merged**